### ISSUE TITLE
Recipe usage

### ISSUE THREAD

**Issue by henry2004y** [MakieOrg/Makie.jl#833]
Hello,

Originally I posted [a question on Discourse](https://discourse.julialang.org/t/makie-recipe-usage/54292), but no one responded. I explained my confusion when using the recipes in Makie. Can you offer me some help in getting this through?

Basically I am asking about two things:
1. How to correctly use recipe for Makie, with a minimal working example? The complete example shown in the [Full Recipes](https://makie.juliaplots.org/stable/recipes.html#Full-recipes-with-the-@recipe-macro) section is a little bit hard to follow for me, even though it actually works.
2. What is the difference between [MakieRecipes.jl](https://github.com/JuliaPlots/MakieRecipes.jl) and the `convert_arguments()` or `@recipe` stuff?

The details are explained in the linked post. Thanks for your response ahead!

**Comment by henry2004y** [comment]
I have seen something in the Makie milestone related to this. Hopefully it can be made easier to use and with better tutorial!

**Comment by jkrumbiegel** [comment]
Hi, I've updated the recipe example a couple days ago, could you have a look and comment here if you still don't understand? Then I could try to address those problems

**Comment by henry2004y** [comment]
Oh I haven't notice that! Yes I will check it soon. Thanks for letting me know! I'll provide some feedback later.

**Comment by henry2004y** [comment]
The full recipe documentation is good. The part I don't quite understand is the type recipes. Similar to the issue described in the discourse thread, I haven't managed to make to work for my type.
```
using Makie

struct MyType
   data::Vector
end

convert_arguments(P::Type{<:AbstractPlot}, x::MyType) = (P,rand(10))

fig = Figure(resolution = (800, 500), backgroundcolor = RGBf0(0.98, 0.98, 0.98))
ax1 = fig[1, 1] = Axis(fig, title = "test")

mydata = MyType(ones(10))
plot!(mydata, color=:red)
fig
```
it returned
```
ERROR: LoadError: Plotting for the arguments (::MyType) not defined. If you want to support those arguments, overload plot!(plot::Plot(MyType,))
```

How to make this work?

**Comment by goretkin** [comment]
>     2\. What is the difference between [MakieRecipes.jl](https://github.com/JuliaPlots/MakieRecipes.jl) and the `convert_arguments()` or `@recipe` stuff?

It appears that `MakieRecipes.jl` is intended (i.e. it isn't finished) to allow recipes defined with https://github.com/JuliaPlots/RecipesBase.jl for the `Plots.jl`world to be re-purposed for use with the `AbstractPlotting.jl` world. 

I think someone might be reasonable confused about the two kinds of recipes within the `JuliaPlots` org, and it might be worth mentioning in [the documentation](http://makie.juliaplots.org/stable/recipes.html) that they (unless I am mistaken) are not immediately interchangeable.

Am I right in understanding that if package `Foo.jl` wants to implement a `Plots.jl`-recipe, `Foo.jl` must depend on `RecipesBase.jl`, whereas to implement a `Makie.jl`-recipe, `Foo.jl` must depend on [`AbstractPlotting.jl`](https://github.com/JuliaPlots/AbstractPlotting.jl/blob/bd426869fa1685c89c847f3821d335474798341e/src/recipes.jl#L124-L140)?

**Comment by henry2004y** [comment]
> It appears that MakieRecipes.jl is intended (i.e. it isn't finished) to allow recipes defined with https://github.com/JuliaPlots/RecipesBase.jl for the Plots.jlworld to be re-purposed for use with the AbstractPlotting.jl world.

This is my impression too. However as a user I tend to be lost in the progress of development. Another issue #893 also mentioned the current failure of the recipe system and I am looking forward to the fixes.

**Comment by ffreyer** [comment]
I'm closing this as outdated/fixed by docs updates
